include/irr/video/IGPUVirtualTexture.h

```diff
@@ -0,0 +1,188 @@
+        {
+            return m_driver->createGPUImageView(std::move(_params));
+        }
+
+        IVideoDriver* m_driver = nullptr;
+    };
+
+public:
+    IGPUVirtualTexture(
+        IVideoDriver* _driver,
+        const base_t::IVTResidentStorage::SCreationParams* _residentStorageParams,
+        uint32_t _residentStorageCount,
+        uint32_t _pgTabSzxy_log2 = 8u,
+        uint32_t _pgTabLayers = 256u,
+        uint32_t _pgSzxy_log2 = 7u,
+        uint32_t _tilePadding = 9u,
+        uint32_t _maxAllocatableTexSz_log2 = 14u
+    ) :
+        base_t(
+            _pgTabSzxy_log2,
+            _pgTabLayers,
+            _pgSzxy_log2,
+            _tilePadding
+        ),
+        m_driver(_driver)
+    {
+        m_pageTable = createPageTable(_pgTabSzxy_log2, _pgTabLayers, _pgSzxy_log2, _maxAllocatableTexSz_log2);
+        initResidentStorage(_residentStorageParams, _residentStorageCount);
+    }
+    IGPUVirtualTexture(IVideoDriver* _driver, asset::IAssetManager* _am, asset::ICPUVirtualTexture* _cpuvt) :
+        base_t(
+            _cpuvt->getPageExtent_log2(),
+            _cpuvt->getPageTable()->getCreationParameters().arrayLayers,
+            _cpuvt->getPageExtent_log2(),
+            _cpuvt->getTilePadding(),
+            false
+            ),
+        m_driver(_driver)
+    {
+        //now copy from CPU counterpart resources that can be shared (i.e. just copy state) between CPU and GPU
+        //and convert to GPU those which can't be "shared": page table and VT resident storages along with their images and views
+
+        m_pageTable = std::move(_driver->getGPUObjectsFromAssets(&_cpuvt->getPageTable(), &_cpuvt->getPageTable()+1)->front());
+
+        m_freePageTableLayerIDs = _cpuvt->getFreePageTableLayersStack();
+
+        memcpy(m_layerToViewIndexMapping->data(), _cpuvt->getLayerToViewIndexMapping().begin(), 
+            _cpuvt->getLayerToViewIndexMapping().size()*sizeof(decltype(m_layerToViewIndexMapping)::pointee::value_type));
+
+        const auto& cpuStorages = _cpuvt->getResidentStorages();
+        for (const auto& pair : cpuStorages)
+        {
+            auto* cpuStorage = static_cast<asset::ICPUVirtualTexture::ICPUVTResidentStorage*>(pair.second.get());
+            const asset::E_FORMAT_CLASS fmtClass = pair.first;
+
+            m_storage.insert({fmtClass, core::make_smart_refctd_ptr<IGPUVTResidentStorage>(_driver, _am, cpuStorage)});
+        }
+
+        m_fsamplers.views = _cpuvt->getFloatViews().size() ? _driver->getGPUObjectsFromAssets(_cpuvt->getFloatViews().begin(), _cpuvt->getFloatViews().end()) : nullptr;
+        m_isamplers.views = _cpuvt->getIntViews().size() ? _driver->getGPUObjectsFromAssets(_cpuvt->getIntViews().begin(), _cpuvt->getIntViews().end()) : nullptr;
+        m_usamplers.views = _cpuvt->getUintViews().size() ? _driver->getGPUObjectsFromAssets(_cpuvt->getUintViews().begin(), _cpuvt->getUintViews().end()) : nullptr;
```

Comment:
Same here as there?
https://github.com/buildaworldnet/IrrlichtBAW/pull/474/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.h&file-filters%5B%5D=No+extension&hide-deleted-files=true#r417947568

Reply:
youre going to kill the existing CPU objects